source/shared/core_stmt.cpp

```diff
@@ -2218,145 +2223,195 @@ void get_field_as_string( _Inout_ sqlsrv_stmt* stmt, _In_ SQLUSMALLINT field_ind
                 if( is_truncated_warning( state ) ) {
 #endif // !_WIN32  
```

Comment:
will we get truncated warning if a client buffer is used, (the data has already been fetched beforehand), would it ever reach this part?
similarly, few lines after this, checking if AE is enabled, can we assume client buffer is not used?